### PR TITLE
NikitK/ Refactor 404 page to Atomic design

### PR DESCRIPTION
**Pull request by NikitK-deriv**
Changes:

-   ...

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [x] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: src/features/pages/404/content/types.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+type TNotAvailableImage = {
+    src: string
+    alt: string
+}
+type TButtonLink = {
+    text: TString
+    url: LinkUrlType
+}
+
+export { TInfoItem, TNotAvailableImage, TButtonLink }
```
Comment (farzin-deriv) [inline src/features/pages/404/content/types.ts:17]: @NikitK-deriv We shouldn't need this, Let TS infer the types as those variables are root level.

**Review by farzin-deriv — COMMENTED** [review]
Nice work 🚀

File: src/features/pages/404/content/types.ts (hunk 1 above)
Comment (NikitK-deriv) [reply src/features/pages/404/content/types.ts:17]: We can only remove TNotAvailableImage and due the other have more specific values than just string
Thanks for note @farzin-deriv  👍

File: src/features/pages/404/content/not-found.container.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+                </Typography.Paragraph>
+                <Button.Primary
+                    type={'button'}
+                    className={styles.not_found_button}
+                    radius={'2x'}
+                    textsize={'medium'}
+                    padding_inline={'8x'}
+                    padding_block={'5x'}
+                >
+                    <Link type={'internal'} url={button_link.url} textcolor={'white'}>
```
Comment (mohsen-deriv) [inline src/features/pages/404/content/not-found.container.tsx:53]:
```suggestion
                    <Link url={button_link.url} textcolor={'white'}>
```
No need to pass the type here, please remove it thanks :pray:

Comment (mohsen-deriv) [reply src/features/pages/404/content/not-found.container.tsx:53]: please update your branch with the base branch, I added the `no_hover` props, it disables the hover effect. with that in place you can remove the related styles from your sass file, thanks :+1:

File: src/features/pages/404/content/not-found.module.scss [hunk 3]
```diff
@@ -0,0 +1,16 @@
+@use 'src/features/styles/theme/theme-mixins' as *;
+@import 'src/features/styles/theme/variables';
+
+.not {
+  &_found {
+    padding: 16rem 0;
```
Comment (mohsen-deriv) [inline src/features/pages/404/content/not-found.module.scss:6]: we should remove this, let's have a call for this, you can handle it with `padding_block` prop, but it gives you `8rem` limit. so let's chat. :pray:

**Review by mohsen-deriv — CHANGES_REQUESTED** [review]

File: src/features/pages/404/content/not-found.container.tsx (hunk 2 above)
Comment (NikitK-deriv) [reply src/features/pages/404/content/not-found.container.tsx:53]: Done 👍 Thanks

File: src/features/pages/404/content/not-found.module.scss (hunk 3 above)
Comment (NikitK-deriv) [reply src/features/pages/404/content/not-found.module.scss]: @mohsen-deriv can we reach our needs via this approach?
https://github.com/binary-com/deriv-com/pull/4445/commits/3dc3acdae81f3e2389f2cce01e7c4c8e178b81a7

**Review by mohsen-deriv — APPROVED** [review]

**Pull request merged**